server_admin/topics/clients/oidc/con-secret-rotation.adoc

```diff
@@ -0,0 +1,28 @@
+[id="con-secret-rotation_{context}"]
+[[_secret_rotation]]
+= Client Secret Rotation
+[role="_abstract"]
+For client with <<_client-credentials, Confidential>> <<_access-type, Access Type>> {project_name} supports the functionality of rotating client secrets through <<_client_policies, Client Policies>>.
+
+The client secrets rotation policy provides greater security in order to alleviate problems such as secret leakage. Once enabled, {project_name} supports up to two concurrently active secrets for each client. The policy manages rotations according to the following settings:
+
+* *Secret expiration:* [seconds] - When the secret is rotated, this is the expiration of time of the new secret. The amount, _in seconds_, added to the secret creation date. Calculated at policy execution time.
+* *Rotated secret expiration:* [seconds] - When secret is rotated, this is the remaining expiration time for the old secret. This value should be always smaller than Secret expiration. When this is set to 0, the old secret will be immediately removed during client rotation. The amount, _in seconds_, added to the secret rotation date. Calculated at policy execution time.
+* *Remaining expiration time for rotation during update:* [seconds] - Time period when an update to a dynamic client should perform client secret rotation. Calculated at policy execution time.
+
+Whenever there is a client secret rotation a new main secret is generated and the old client main secret becomes the secondary secret with a new expiration date.
+
+== Rules for client secret rotation
+Rotations do not occur automatically or through a background process. In order to perform the rotation, an update action is required on the client, either through the {project_name} Admin Console UI or Admin REST API. When invoking a client update action, secret rotation occurs according to the rules:
```

Comment:
This is true in case that OIDC Dynamic client update request is used. But for admin REST API, the executor is triggered when new client is created or when client secret is re-generated (EG. Admin clicks on the "Re-generate secret" ). Is it correct? So maybe this part can be slightly re-formulated to mention that "Re-generate secret" is going to check client secret rotation instead of the client update in case of admin REST API?

Reply:
Sorry @mposolda, I don't think I quite understand what you mean. In this paragraph I would just like to mention that the rotation can be done either through the Admin Console or via REST. I didn't think the detail of the executioner's performance made any difference.

Maybe mention that the "Re-generate secret" button performs an update on the client?

Reply:
> Maybe mention that the "Re-generate secret" button performs an update on the client?

Yes, exactly. That will be probably good. As for the admin console, the "update of the client" implies to me that I just click on the client, update some settings of the client in the tab "Details" and click "Save" . Which won't help in this case if I understand correctly?

Reply:
@mposolda You are right.

Reply:
Updated